### PR TITLE
fix(samples): Update lightweight_component, output_a_directory samples to v2 pipelines

### PR DESCRIPTION
**Pull request by junggil**
**Description of your changes:**
Update lightweight_component, output_a_directory samples to v2 pipelines

**Checklist:**
- [v] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: samples/core/lightweight_component/lightweight_component.ipynb [hunk 1]
```diff
@@ -30,17 +32,17 @@
    "outputs": [],
    "source": [
     "# Install the SDK\n",
-    "!pip3 install 'kfp>=0.1.31.2' --quiet"
+    "!pip install 'kfp>=2' --quiet"
```
Comment (chensun) [inline samples/core/lightweight_component/lightweight_component.ipynb:35]: nit: how about `kfp==2.*`?

File: samples/core/lightweight_component/lightweight_component.ipynb [hunk 2]
```diff
@@ -2,26 +2,28 @@
...
     "* The function can only import packages that are available in the base image. If you need to import a package that's not available you can try to find a container image that already includes the required packages. (As a workaround you can use the module subprocess to run pip install for the required package. There is an example below in my_divmod function.)\n",
     "* If the function operates on numbers, the parameters need to have type hints. Supported types are ```[int, float, bool]```. Everything else is passed as string.\n",
-    "* To build a component with multiple output values, use the typing.NamedTuple type hint syntax: ```NamedTuple('MyFunctionOutputs', [('output_name_1', type), ('output_name_2', float)])```"
+    "* To build a component with multiple output values, use the typing.NamedTuple type hint syntax: ```NamedTuple('MyFunctionOutputs', output_name_1=type), output_name_2=type, ...)```. You can access a named output using ```.output['<output-key>']```."
```
Comment (chensun) [inline samples/core/lightweight_component/lightweight_component.ipynb:21]: `.output['<output-key>']` should be `.outputs['<output-key>']` (plural) instead.

Comment (junggil) [reply samples/core/lightweight_component/lightweight_component.ipynb:21]: Thanks, that makes sense. I do agree.  
Note that the public documents[1][2], we described as singular form and I took the notation as it is. 

Let me fix the documentation as well. 

[1] https://www.kubeflow.org/docs/components/pipelines/v2/pipelines/pipeline-basics/#data-passing-and-task-dependencies
[2] https://www.kubeflow.org/docs/components/pipelines/v2/data-types/parameters/#multiple-output-parameters

File: samples/core/lightweight_component/lightweight_component.ipynb (hunk 1 above)
Comment (junggil) [reply samples/core/lightweight_component/lightweight_component.ipynb]: Done

File: samples/core/lightweight_component/lightweight_component.ipynb (hunk 2 above)
Comment (junggil) [reply samples/core/lightweight_component/lightweight_component.ipynb]: Done

**Comment by junggil** [comment]
Note that just added a doc PR as well: https://github.com/kubeflow/website/pull/3578

**Review by chensun — APPROVED** [review]
/lgtm

**Review by chensun — APPROVED** [review]
/lgtm

**Pull request closed without merging**